yt_dlp/extractor/yle_areena.py

```diff
@@ -0,0 +1,41 @@
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+)
+
+
+class YleAreenaIE(InfoExtractor):
+    _VALID_URL = r'https?://areena\.yle\.fi/(?P<id>[0-9-]+)'
+    _TESTS = [{
+        'url': 'https://areena.yle.fi/1-4371942',
+        'md5': '932edda0ecf5dfd6423804182d32f8ac',
+        'info_dict': {
+            'id': '0_a3tjk92c',
+            'ext': 'mp4',
+            'title': 'K1, J2: Pouchit | Modernit miehet',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader_id': 'ovp@yle.fi',
+            'duration': 1435,
+            'view_count': int,
+            'upload_date': '20181204',
+            'timestamp': 1543916210,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        video_data_url = f'https://player.api.yle.fi/v1/preview/{video_id}.json?app_id=player_static_prod&app_key=8930d72170e48303cf5f3867780d549b'
```

Comment:
are u sure the app_key is static?

Reply:
Can't be sure that it won't change in the future. But it does seem static for now. I get same app_key in non-private and private sessions of several browsers. Also it was proposed in origin issue by another person.

Reply:
Nope, it's not static. After several reloads it changed to `v9No1mV0omg2BppmDkmDL6tGKw1pRFZt`. So it can change. But old one is still working.
I've found `app_key` in file https://areena-web-items.ylestatic.fi/3.0.2892/_next/static/chunks/pages/_app-b59cbfac9f6f0c9f.js. Any suggestions how to get it from there?

Reply:
I've found that web application downloads file https://player-v2.yle.fi/embed.js. Inside I've found code which produces links to download more js-files.
```
a.u=function(t){return(37===t?"MediaSDK":t)+"."+{37:"3a400b149dc9743bf75f",369:"b2fdad59226016eaf4f9",899:"d455e7e61b1f47354798"}[t]+".js"}
```
One of the files https://player-v2.yle.fi/369.b2fdad59226016eaf4f9.js contains app_key
```
app_id=player_static_prod&app_key=8930d72170e48303cf5f3867780d549b
```

It seems simple to parse app_key, but I have no idea how to reliably to get link from https://player-v2.yle.fi/embed.js

Reply:
"static for now" is good enough